### PR TITLE
Theme split

### PR DESCRIPTION
**Pull request by cwolf10**
Splitting SAM theme into the base USWDS and SAM theme. USWDS will apply standard USWDS css and color scheme as we as our custom elements and a few additional variables required by those elements. SAM then overrides those variables set in USWDS as needed to get the desired look. For this to be imported into SDS, sds-styles.scss needs to be updated to point to sam.scss in sam-styles.

To only apply USWDS, in sam.scss remove the line importing the sam theme. To add a different theme, update the same line to point at the desired theme's _style.scss folder.

### CODE REVIEW

File: src/stylesheets/uswds.scss [hunk 1]
```diff
@@ -0,0 +1,10 @@
+
+/*******************************
+          SDS THEME
+*******************************/
+
+@import './themes/USWDS/required-vars';
```
[the remark this answers is not in the document]
Comment (Bibek-Regmi) [reply src/stylesheets/uswds.scss:6]: Ah I see what you mean. That's actually a bit annoying that USWDS did not define default values for them. Could you confirm that those themes are also being used in sam styles?

[the remark this answers is not in the document]
Comment (cwolf10) [reply src/stylesheets/uswds.scss:6]: Yeah the first 4 vars are being used in sam-styles. However all but semi-bold are only being used once. Semibold gets used almost 30 times in sam-styles.

**Comment by cwolf10** [comment]
Yeah that is tied to the theme switching. Had the same issue on that branch, never did find out why federalist did not work with it. I'll spend a bit of time trying to isolate, in the mean time if you want to play around with the branch, just pull locally.

**Comment by Bibek-Regmi** [comment]
Yes, it seems to work fine locally. After messing around a bit, I noticed that the USWDS theme also includes elements from sam-styles, which causes components to look very off. But we cannot really remove sam-styles components when displaying USWDS theme because there are components in sam-styles that don't exist in USWDS, so those styles need to depend on styles from Sam Styles. There's generally 3 cases:
- Component exists in USWDS, but not Sam styles, which shouldn't need any work as they don't exist in sam styles
- Components exists in both USWDS and Sam Styles - These are the alerts / buttons / accordions, and there's work there to align the classes we have with USWDS. This was the case Sam was working on
- Components that do not exist in USWDS, but do in Sam Styles - This forces us to incorporate sam styles component changes on USWDS theme, because otherwise, those components would have no styles on pure uswds. 

I think to handle the 3rd case, the most straightforward approach would be to add in components of that case (exists in sam but not uswds) specifically into the uswds theme. We could maybe even separate out components in sam styles into USWDS overrides vs Sam styles. Let me know your thoughts

**Comment by cwolf10** [comment]
By very off you are comparing the components with USWDS applied to the components from the USWDS site?

I did not split the components out in this case after speaking with Dave who suggested that they be imported for both themes and have the theme vars be what is different between them.

I did take a look at splitting them off, it is possible to do, however it does complicate the process for setting up and applying the theme. Basically it ends up being a multi-layer process where first the applied theme, in this case SAM, needs to be imported then USWDS so that anything which has not been overridden by SAM has a value. After that you'd need to import both set of elements, followed by both sets of components.

There is also a fair bit of interdependency between the elements and components so this also assumes you can break them apart in so that the required mixins are present when needed.

**Comment by Bibek-Regmi** [comment]
Yeah - looking at just buttons for USWDS - it might just be due to the stark contrast between uswds' secondary and our secondary colors. The text color for those buttons end up being red because sam styles uses a dark shade of secondary for primary button text color. This brings up dependency between the two. Ideally, uswds should override whatever sam styles does for uswds theme. Maybe just switching the ordering of imports such that sam styles components/elements are imported first, then uswds can resolve this

**Comment by cwolf10** [comment]
Yeah need to follow up with Dave on that, probably one of the scenarios he was expecting to rear its head and make this effort tricker. BTW the switcher is fixed in federalist.

**Pull request closed without merging**